src/main/scala/com/github/unchama/seichiassist/subsystems/seasonalevents/anniversary/AnniversaryListener.scala

```diff
@@ -49,9 +55,67 @@ class AnniversaryListener(implicit effectEnvironment: EffectEnvironment,
         grantItemStacksEffect(mineHead),
         MessageEffect(s"${BLUE}ギガンティック☆整地鯖${ANNIVERSARY_COUNT}周年の記念品を入手しました。"),
         FocusedSoundEffect(Sound.BLOCK_ANVIL_PLACE, 1.0f, 1.0f),
-        UnfocusedEffect{playerData.anniversary = false}
+        UnfocusedEffect {
+          playerData.anniversary = false
+        }
       ),
       s"${ANNIVERSARY_COUNT}周年記念ヘッドを付与する"
     )
   }
+
+  @EventHandler(ignoreCancelled = true)
+  def onPlayerPlaceSapling(event: BlockPlaceEvent): Unit = {
+    if (!isStrangeSapling(event.getItemInHand)) return
+
+    val placedBlock = event.getBlock
+    // 苗木をなくす
+    placedBlock.setType(Material.AIR)
+    val location = placedBlock.getLocation
+    // オークの木を生やす
+    location.getWorld.generateTree(location, TreeType.TREE)
+
+    val breaks = new Breaks
+    // Y座標を下に動かして（木の上方から）オークの木の頂点を探し、そのブロックを置き換えて、ループを抜ける
+    breaks.breakable {
+      for (relY <- 10 to 0 by -1) {
+        val block = placedBlock.getRelative(0, relY, 0)
+
+        if (block.getType == Material.LOG || block.getType == Material.LEAVES) {
+          replaceBlockOnTreeTop(location.getBlock, event.getPlayer.getName)
+          breaks.break
+        }
+      }
+    }
```

Comment:
```suggestion
    (10 to 0 by -1)
      .map(placedBlock.getRelative(0, _, 0))
      .map(_.getType)
      .find {
        case Material.LOG | Material.LEAVES => true
        case _ => false
      }
      .foreach { _ =>
        replaceBlockOnTreeTop(location.getBlock, event.getPlayer.getName)
      }
```
的なコードの方がパフォーマンス的には良さげ